### PR TITLE
Adding FCM pieces

### PR DESCRIPTION
**Pull request by secondsun**
Do not merge until fh-android-sdk 3.1.0 hits maven central

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -15,6 +15,7 @@ buildscript {
 allprojects {
     repositories {
         jcenter()
+        mavenLocal()
         maven { url "https://github.com/feedhenry/fh-android-sdk-repository/raw/master/releases" }
```
Comment (danielpassos) [inline build.gradle]: We should remove `mavenLocal()` and the old `fh-android-sdk-repository`?

File: app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -40,6 +40,10 @@
         android:theme="@style/MyTheme.Base"
         tools:replace="android:icon">
 
+        <meta-data
+            android:name="DEFAULT_MESSAGE_HANDLER_KEY"
+            android:value="org.jboss.aerogear.unifiedpush.helloworld.handler.NotificationBarMessageHandler" />
```
Comment (danielpassos) [inline app/src/main/AndroidManifest.xml]: s/org.jboss.aerogear.unifiedpush.helloworld.handler/org.feedhenry.pushstarter.handler

File: build.gradle (hunk 1 above)
Comment (matzew) [inline build.gradle]: let's do that once we do have something on maven central, not yet the case

File: app/src/main/AndroidManifest.xml (hunk 2 above)
Comment (matzew) [inline app/src/main/AndroidManifest.xml]: 👍 good catch 🐱

Comment (secondsun) [inline app/src/main/AndroidManifest.xml]: Fixed

File: app/build.gradle [hunk 3]
```diff
@@ -34,7 +34,9 @@ license {
 }
 
 dependencies {
-    compile 'com.feedhenry:fh-android-sdk:3.0.0'
+    compile 'com.feedhenry:fh-android-sdk:3.1.0-SNAPSHOT'
     compile 'com.android.support:appcompat-v7:23.1.1'
-    compile 'org.jboss.aerogear:aerogear-android-push:2.2.2'
+    compile 'org.jboss.aerogear:aerogear-android-push:4.0.0-SNAPSHOT'
```
Comment (matzew) [inline app/build.gradle]: @secondsun can you update this line to be 4.0.0 ?

File: app/build.gradle [hunk 4]
```diff
@@ -34,7 +34,9 @@ license {
 }
 
 dependencies {
-    compile 'com.feedhenry:fh-android-sdk:3.0.0'
+    compile 'com.feedhenry:fh-android-sdk:3.1.0-SNAPSHOT'
```
Comment (matzew) [inline app/build.gradle]: and this one too? since we have staging repo, it should not be -SNAPSHOT

**Comment by secondsun** [comment]
@matzew Should be fixed now

**Comment by matzew** [comment]
I was able to build this on the build-farm, after adding a `google-services.json` file, with the proper config (e.g. `org.feedhenry.pushstarter` package)

**Comment by matzew** [comment]
👍

**Comment by matzew** [comment]
latest staging repo: https://repository.jboss.org/nexus/content/repositories/jboss_thirdparty_releases_staging_profile-1552/

**Comment by danielpassos** [comment]
👍 to merge it when available on Maven central

**Pull request merged**